app/controllers/admin/payments_controller.rb

```diff
@@ -0,0 +1,31 @@
+module Admin
+  class PaymentsController < BaseAdminController
+    before_action :ensure_service_operator
+    before_action :find_payroll_run
+    before_action :find_payment
+
+    def delete
+    end
+
+    def destroy
+      if @payroll_run.confirmation_report_uploaded_by.nil?
```

Comment:
Is there an undefined concept/idea here that we could encapsulate in a method on `PayrollRun` to better communicate what is going on here?

Reply:
👍 I've added a method to `PayrollRun`:

```
def confirmation_report_uploaded?
  confirmation_report_uploaded_by.present?
end
```